components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,586 @@
+        Assert.assertEquals(creatingAction.getName(), action.getName());
+        Assert.assertNull(null, action.getDescription());
+        Assert.assertEquals(creatingAction.getType(), action.getType());
+        Assert.assertEquals(creatingAction.getStatus(), action.getStatus());
+        Assert.assertEquals(creatingAction.getEndpoint().getUri(), action.getEndpoint().getUri());
+        Assert.assertEquals(creatingAction.getEndpoint().getAuthentication().getType(),
+                action.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 7)
+    public void testUpdateAction() throws ActionMgtException {
+
+        Action updatingAction = buildMockUpdatingActionWithBasicAuth(
+                "Pre Issue Access Token",
+                "To configure pre issue access token",
+                "https://sample.com",
+                "updatingadmin",
+                "updatingadmin");
+        Action result = daoImpl.updateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), updatingAction, action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(updatingAction.getName(), result.getName());
+        Assert.assertEquals(updatingAction.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(updatingAction.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(
+                updatingAction.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType()
+        );
+        action = result;
+    }
+
+    @Test(priority = 8)
+    public void testUpdateActionWithoutNameAndDescription() throws ActionMgtException {
+
+        Action updatingAction = buildMockUpdatingActionWithBasicAuth(
+                null,
+                null,
+                "https://sample.com",
+                "updatingadmin",
+                "updatingadmin");
+        Action result = daoImpl.updateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), updatingAction, action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(action.getName(), result.getName());
+        Assert.assertEquals(action.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(updatingAction.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(
+                updatingAction.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType()
+                           );
+    }
+
+    @Test(priority = 9)
+    public void testUpdateActionWithoutEndpointConfiguration() throws ActionMgtException {
+
```

Comment:
Shouldn't the intention of this method to assert the exception ?

Reply:
ActionManagementDAOImpl class lets user to update an action without giving an endpoint . Therefore this test case is for a successful one,  not for an exception.